### DISCUSSION TITLE
Multiple Variants Per Component

### DISCUSSION THREAD

**Discussion by sam-cuboh** [mantinedev/mantine#7139] [Feature requests]
I was wondering if it would be possible to add support for multiple variants (similar to how classNames work). for example you could have something like

```
<Anchor variant="caption link" />
```

which you could then target and style 

```
&[data-variant='caption'] {
    text-transform: uppercase;
  }

&[data-variant='link'] {
    text-decoration: underline;
  }
```

**Comment by rtivital** [comment]
Create `caption-link` variant instead

**Reply by sam-cuboh** [reply]
That is what I am currently doing. however, we have many overlapping and complex variants, so this solution would allow you to layer and pick and choose variants for different cases cleanly.

**Reply by rtivital** [reply]
In this case, you can implement that with `data-*` attributes, like `data-link` and `data-caption`

**Reply by sam-cuboh** [reply]
sounds good thanks! I think the above solution would be a lot cleaner, but this will definitely do the trick.
